executor/runner/kernel_logs.go

```diff
@@ -0,0 +1,118 @@
+package runner
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+func readCgroupFile(pid int) ([]string, error) {
+	rawCgroups, err := os.Open(fmt.Sprintf("/proc/%d/cgroup", pid))
+	if err != nil {
+		return nil, err
+	}
+	defer rawCgroups.Close()
+
+	lines := []string{}
+	scanner := bufio.NewScanner(rawCgroups)
+	for scanner.Scan() {
+		line := scanner.Text()
+		// ignore last empty line
+		if len(line) == 0 {
+			break
+		}
+		lines = append(lines, line)
+	}
+
+	return lines, nil
+}
+
+var errNotTitusTask = fmt.Errorf("Not a titus task")
+
+func getControllerAndTaskid(line string) (string, string, error) {
+	fields := strings.Split(line, ":")
+	if len(fields) != 3 {
+		return "", "", fmt.Errorf("strange /proc/pid/cgroup line %s\n", line)
+	}
+
+	re := regexp.MustCompile("titus-executor@default__([0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}).service")
+
+	matches := re.FindStringSubmatch(fields[2])
+	if len(matches) != 2 {
+		return "", "", errNotTitusTask
+	}
+
+	return fields[1], matches[1], nil
+}
+
+func findTitusTaskIdFromPid(pid int) (string, error) {
+	controllers, err := readCgroupFile(pid)
+	if err != nil {
+		return "", err
+	}
+
+	if len(controllers) == 1 {
+		// we're in the "unified"/cgroupv2 heirarchy
+		_, id, err := getControllerAndTaskid(controllers[0])
+		return id, err
+	} else {
+		// pick our favorite controller (i.e. the one we think that
+		// titus is most likely to keep using)... cpuset?
+		for _, line := range controllers {
+			controller, id, err := getControllerAndTaskid(line)
+			if err != nil {
+				return "", err
+			}
+
+			if controller == "cpuset" {
+				return id, nil
+			}
+		}
+
+		return "", fmt.Errorf("couldn't find cpuset controller for pid %d\n", pid)
+	}
+}
+
+func handleKernelLogLine(line string, task Task) (string, error) {
```

Comment:
This function only handles `blocked for more than` log lines.
I think it would be better named `parseBlockedTaskKernelLogLine`, until it handles more than that